src/lib.rs

```diff
@@ -29,15 +30,54 @@ mod dlmalloc;
 #[cfg(all(feature = "global", not(test)))]
 mod global;
 
+/// A platform interface
+pub unsafe trait System: Send {
```

Comment:
To bikeshed a bit, the documentation and wording here are used somewhat interchangeably (e.g. this trait's documentation says "platform" where the trait says "system"). Perhaps the trait could be `Allocator` and the default struct could be `System` (to mirror `std::alloc` a bit).